src/protocol.rs

```diff
@@ -38,10 +40,32 @@ pub enum ProtocolType {
     Broadcast,
 }
 
-/// A `Participant` is the driver for a party executing a subprotocol of the
-/// threshold signing protocol.
+/// The driver for a party executing a sub-protocol of the threshold signing
+/// protocol.
+///
+/// A given [`Participant`] participates in an execution of one of several
+/// sub-protocols required for threshold signing. The core functionality of
+/// [`Participant`] is captured in the
+/// [`process_single_message`](Participant::process_single_message) method: it
+/// takes as input a [`Message`] and outputs a tuple containing the
+/// participant's output alongside a list of messages to process.
+///
+/// # ⚠️ Storage requirements ⚠️
+/// It is up to the calling application to persist outputs used by the
+/// participant. In addition, some of the outputs are private to the
+/// participant, and **these must be stored securely by the calling
+/// application**. Which outputs require secure storage is documented by each
+/// protocol type, under the "Storage requirements" heading:
+/// [`KeygenParticipant`](crate::KeygenParticipant),
+/// [`AuxInfoParticipant`](crate::AuxInfoParticipant), and
+/// [`PresignParticipant`](crate::PresignParticipant). In addition, some outputs
+/// must only be used once and then discarded. These are documented as necessary
+/// under the "Lifetime requirements" heading in the aforementioned types.
 ///
-/// Each participant has an inbox which can contain messages.
+/// ## Requirements of secure storage
+/// Any external secure storage must be able to achieve the following two
+/// requirements: (1) the ability to store data in encrypted form, and (2) the
+/// ability to securely erase data from storage.
```

Comment:
todo: The other property @indomitableSwan [has in the spec](https://github.com/boltlabs-inc/LK-crypto-protocols/blob/develop/basic_implementation_guidance.md#desired-database-properties) is freshness -- the ability to prove that retrieved data is from the most recent storage state.